sky/packages/sky/lib/src/animation/timeline.dart

```diff
@@ -5,17 +5,89 @@
 import 'dart:async';
 
 import 'package:newton/newton.dart';
-import 'package:sky/src/animation/curves.dart';
 import 'package:sky/src/animation/animated_value.dart';
-import 'package:sky/src/animation/animated_simulation.dart';
+import 'package:sky/src/animation/curves.dart';
+import 'package:sky/src/animation/scheduler.dart';
+
+typedef _TickerCallback(Duration elapsed);
+
+/// Calls its callback once per animation frame
+class _Ticker {
```

Comment:
This class (_Ticker) actually seems like a legitimate primitive. Maybe it should be in the scheduler.dart file or in its own ticker. dart file. It feels like a lower level than the rest of the animation system.